src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs

```diff
@@ -1616,7 +1616,13 @@ public void EnsureCallingThePropertySetter()
             Assert.Equal(2, options.ParsedBlacklist.Count); // should be initialized when calling the options.Blacklist setter.
 
             Assert.Equal(401, options.HttpStatusCode); // exists in configuration and properly sets the property
-            Assert.Equal(2, options.OtherCode); // doesn't exist in configuration. the setter sets default value '2'
+#if BUILDING_SOURCE_GENERATOR_TESTS
+            // Setter not called if there's no matching configuration value.
+            Assert.Equal(0, options.OtherCode);
```

Comment:
@tarekgh this doesn't comply with the fix https://github.com/dotnet/runtime/pull/80438. I believe this is the right behavior and we can implement it since the source generator is new.

Reply:
It is important to have the source gen implement this behavior ensuring the setter is called. Without this behavior we were getting a lot of regressions if this behavior is not there. Can you please track it to fix it in the source gen?